sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -495,7 +493,8 @@ private static boolean maybeUsePortResources(Collection<String> networkNames) {
         return networkNames.size() == 0;  // if we have no networks, we want to use port resources
     }
 
-    private static ResourceSpec from(String role, String principal, WriteOnceLinkedHashMap<String, RawPort> rawPorts) {
+    private static ResourceSpec from(String role, String principal, WriteOnceLinkedHashMap<String, RawPort> rawPorts,
+                                     Collection<String> networkNames) {
```

Comment:
Yes, this is intentional. I need to have all of the network names at the PortSpec level to pass information along to resourceRequirements.